README.md

```diff
@@ -1,15 +1,126 @@
+
+`(u)` means the command is as-yet unimplemented.
+
+| Command   | Use                                                                 |
+|-----------|---------------------------------------------------------------------|
+| agent (u) | Starts a Holochain node as an agent                                 |
+| generate  | Generates a new zome and scaffolds the given capabilities           |
+| init      | Initializes a new Holochain app at the given directory              |
+| package   | Builds the current Holochain app into a `.hcpkg` file               |
+| unpack    | Unpacks a Holochain bundle into it's original file system structure |
+| web (u)   | Starts a web server for the current Holochain app                   |
+
+## How To Get Started Building An App
+
+In your terminal, change directories to one where you wish to initialize a new Holochain app.
+Run the following, replacing `your_app_name` with your actual app name:
+```shell
+$ hcdev init your_app_name
+$ cd your_app_name
+```
+
+We now have the empty shell of a Holochain app. From here, we will want to generate at least one Zome.
+To do this, run the following, replacing `your_zome_name` with a name related to the functionality you wish to develop. For example: `users`.
+```shell
+$ hcdev generate zomes/your_zome_name rust
+```
+
+Currently, Rust is the only language that the tool will scaffold a Zome for. In the future, other languages will be included. In the command above, we declared `rust` just to be explicit, even though it's the default language.
+
+What this did is generate a new folder under `zomes` called `users`. Here is the folder structure of it.
+- users
+  - code
+    - src
+      - main.rs
+    - .build
+    - Cargo.toml
+  - zome.json
+
+So in a given Zome we have two things:
+1. a JSON file, `zome.json`, which defines and configures the Zome
+2. a `code` folder, which can be compiled into a single `WASM` file with the code for this Zome
+
+In order for Holochain to run your app, you have to build your code into a single packaged file. Those instructions follow.
+
+## What are .hcpkg files?
+
+A Holochain app can be fully contained in a file known as a `.hcpkg` file.
+It is a JSON file, with a particular structure that Holochain can understand, and execute.
+
+This is an unusual JSON file, it is part configuration, and part executable.
+
+The configuration part comes from the `json` files that are throughout your app. One at the top level for the application (`app.json`) and one for each Zome. Ultimately, these get stitched together into a single tree structure in the `.hcpkg` file.
+
+The executable part comes from having embedded Base64 encoded WebAssembly code in the file. *What does that mean?* [WebAssembly](https://webassembly.org/) is a fast and secure low-level language.
+Rather than storing the code in its raw format, Holochain expects the code to be [encoded using Base64](https://en.wikipedia.org/wiki/Base64) , so that its as compact as possible.
```

Comment:
doesn't base64 make things ~4/3x bigger not smaller?

https://stackoverflow.com/questions/4715415/base64-what-is-the-worst-possible-increase-in-space-usage

i assume the base64 is to make the string as portable/safe/legible as possible vs. raw wasm bytes

Reply:
Ah, yes, I hadn't realized that. Thanks David